docs/pydoc/renderers.py

```diff
@@ -19,6 +19,42 @@
 
 """
 
+class HaystackMarkdownRenderer(MarkdownRenderer):
+    """
+    Custom Markdown renderer heavily based on the `MarkdownRenderer`
+    """
+    def _render_object(self, fp, level, obj):
+        """
+        This is where docstrings for a certain object are processed,
+        we need to override it in order to better manage new lines.
+        """
+        if not isinstance(obj, docspec.Module) or self.render_module_header:
+            self._render_header(fp, level, obj)
+
+        render_view_source = not isinstance(obj, (docspec.Module, docspec.Variable))
+
+        if render_view_source:
+            url = self.source_linker.get_source_url(obj) if self.source_linker else None
+            source_string = self.source_format.replace("{url}", str(url)) if url else None
+            if source_string and self.source_position == "before signature":
+                fp.write(source_string + "\n\n")
+
+        self._render_signature_block(fp, obj)
+
+        if render_view_source:
+            if source_string and self.source_position == "after signature":
+                fp.write(source_string + "\n\n")
+
+        if obj.docstring:
+            docstring = html.escape(obj.docstring.content) if self.escape_html_in_docstring else obj.docstring.content
+            docstring = docstring.replace("**Arguments**", "\n\n**Arguments**")
+            docstring = docstring.replace("**Returns**", "\n\n**Returns**")
+            docstring = docstring.replace("**Raises**", "\n\n**Raises**")
+            lines = docstring.split("\n\n")
```

Comment:
I think the real fix is this one, splitting by `\n\n` instead of just `\n`, the problem of `Arguments` not being properly surrounded by newlines is different (and new?). We should check if an issue already exists in pydoc-markdown's repo and if not opening one. Then we can reference the issue in a comment about why we're `replace`ing.

Reply:
> the problem of `Arguments` not being properly surrounded by newlines is different (and new?). We should check if an issue already exists in pydoc-markdown's repo and if not opening one. Then we can reference the issue in a comment about why we're `replace`ing.

My understanding after playing around with the code:
- The sections (Arguments, Returns, Raises etc) seem to be defined by the [processors](https://github.com/NiklasRosenstein/pydoc-markdown/tree/3aa2fa94f514ee827992e4145a23e1260b5a4efd/src/pydoc_markdown/contrib/processors)
- We seem to be using the `smart` processor
- I attempted to customize a processor in this file here too. I was unsuccessful in trying to fix this 'Arguments' issue like that

But most importantly here is my understanding, I may be wront:
- The issue isn't new, it just was not an issue when we were splitting by `\n` only
- I think the generated 'sections' all started with a single `\n` and now that we want to split by `\n\n` they're not getting placed on a new line :(
- This is what I tried to fix by implementing a custom processor in our code. But I was terribly unsuccessful and I think it's because we make use of the `smart` one. Some of the generated markdown sections were fixed but some were not. 
I am stuck at this final step, which is why I just hardcoded to replaces in here 😢 - Any help would be appreciated. Even in understanding what's going on. You could test it out by simply removing the replaces.